src/main/java/duke/command/CommandLibrary.java

```diff
@@ -0,0 +1,69 @@
+package duke.command;
+
+import java.util.List;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+class CommandLibrary {
+    static final CommandExecutable DEADLINE_COMMAND = (taskList, ui, args) -> {
+        String description = args[0].trim();
+        if (args.length != 2 || "".equals(description)) {
+            throw DukeException.Errors.DEADLINE_BAD_FORMAT.create();
+        }
+        String time = args[1];
+        ui.systemMessage(taskList.add(new Deadline(description, time)));
+    };
+    static final CommandExecutable EVENT_COMMAND = (taskList, ui, args) -> {
+        String description = args[0].trim();
+        if (args.length != 2 || "".equals(description)) {
+            throw DukeException.Errors.EVENT_BAD_FORMAT.create();
+        }
+        String time = args[1];
+        ui.systemMessage(taskList.add(new Event(description, time)));
+    };
+    static final CommandExecutable TODO_COMMAND = (taskList, ui, args) -> {
+        String description = args[0];
+        if ("".equals(description)) {
+            throw DukeException.Errors.TODO_EMPTY_DESCRIPTION.create();
+        }
+        ui.systemMessage(taskList.add(new Todo(description)));
+    };
+    static final CommandExecutable DELETE_COMMAND = (taskList, ui, args) -> {
+        int idx = Integer.parseInt(args[0]);

```

Comment:
Perhaps naming the variable as 'index' is clearer?

Reply:
Thanks for pointing this out, didn't notice that I was using shorthands.